tests/cli/set/test_sample.py

```diff
@@ -194,6 +194,38 @@ def test_priority_number(cli_runner: CliRunner, base_context: CGConfig, base_sto
     assert base_context.lims_api.get_updated_sample_value() == sample_obj.priority_human
 
 
+def test_sample_comment(cli_runner: CliRunner, base_context: CGConfig, base_store: Store, helpers):
+    # GIVEN a database with a sample without comment
+    sample_obj = helpers.add_sample(base_store, gender="female")
+    key = "comment"
+
+    # WHEN setting key on sample to new_value
+    new_value = "test comment"
+    result = cli_runner.invoke(
+        sample, [sample_obj.internal_id, "-kv", key, new_value, "-y"], obj=base_context
+    )
+
+    # THEN it should have new_value as attribute key on the sample and in LIMS
+    assert result.exit_code == SUCCESS
+    assert getattr(sample_obj, key).endswith(new_value)
+    assert base_context.lims_api.get_updated_sample_key() == key
+    assert base_context.lims_api.get_updated_sample_value() == new_value
+
+    # WHEN setting key again on sample this time to newer_value
+    newer_value = "another test comment"
+    result = cli_runner.invoke(
+        sample, [sample_obj.internal_id, "-kv", key, newer_value, "-y"], obj=base_context
+    )
+
+    # THEN it should have newer_value above new_value as attribute key on the sample and in LIMS
+    comments = getattr(sample_obj, key).split("\n")
+    assert result.exit_code == SUCCESS
+    assert comments[0].endswith(newer_value)
+    assert comments[1].endswith(new_value)
+    assert base_context.lims_api.get_updated_sample_key() == key
+    assert base_context.lims_api.get_updated_sample_value() == newer_value
```

Comment:
I think this test should be split into two tests to get independent test-cases, one that tests setting a comment on a sample without comment and one test that tests setting comment on a sample with an existing comment

Reply:
> I think this test should be split into two tests to get independent test-cases, one that tests setting a comment on a sample without comment and one test that tests setting comment on a sample with an existing comment

I agree. I'll split the test into two separate test cases.